CesiumGS/cesium
# Update prettier to v3

# Description

Updates prettier to v3.3.0 which crosses the major version 3 [breaking changes](https://prettier.io/blog/2023/07/05/3.0.0.html#breaking-changes)

This is a _very_ big pr touching nearly all our source files but all the changes are basically identical.

Key changes:

- Trailing commas are now always included (a very good change IMO), this is a majority of the changes
- greatly improved line wrapping logic for readability (see below)
- prefer lowercase `doctype` over `DOCTYPE`

```
// before 3.0 it preferred parens and brackets
const tileset = await Cesium.Cesium3DTileset.fromIonAssetId(
  1
);

// after 3.0 it actually somewhat uses context, much more readable imo
const tileset =
  await Cesium.Cesium3DTileset.fromIonAssetId(1);

```

I also added a new `.prettierrc` specifically for the Sandcastle Gallery to allow a slightly wider line with to account for the indentation that's removed when sandcastles are actually rendered. This should produce code that matches prettier's default 80 when in the browser editor.


When we decide to merge this I'll add some new git tags and post a guide on how to update other branches easier.







## Issue number and link

Fixes https://github.com/CesiumGS/cesium/issues/12172





## Testing plan

- Run `npm run prettier` and make sure there are no changed files
    - or just run `npm run prettier-check` and make sure it doesn't fail
- make sure specs still pass
- functionally everything should be the same so primarily review the files themselves.




# Author checklist
